Code/Tools/AssetProcessor/native/tests/UuidManagerTests.cpp

```diff
@@ -319,17 +319,45 @@ namespace UnitTests
 
     TEST_F(UuidManagerTests, TwoFilesWithSameRelativePath_EnabledType_ReturnsDifferentUuid)
     {
-        static constexpr const char* FileA = "c:/somepath/folderA/mockfile.txt";
-        static constexpr const char* FileB = "c:/somepath/folderB/mockfile.txt";
+        static constexpr AZ::IO::FixedMaxPath FileA = "c:/somepath/folderA/mockfile.txt"; // txt files are enabled
+        static constexpr AZ::IO::FixedMaxPath FileB = "c:/somepath/folderB/mockfile.txt";
 
         MakeFile(FileA);
         MakeFile(FileB);
 
-        auto uuidA = m_uuidInterface->GetUuid(AssetProcessor::SourceAssetReference(1, "c:/somepath/folderA", "mockfile.txt"));
-        auto uuidB = m_uuidInterface->GetUuid(AssetProcessor::SourceAssetReference(2, "c:/somepath/folderB", "mockfile.txt"));
+        auto uuidA = m_uuidInterface->GetUuid(AssetProcessor::SourceAssetReference(1, FileA.ParentPath(), FileA.Filename()));
+        auto uuidB = m_uuidInterface->GetUuid(AssetProcessor::SourceAssetReference(2, FileB.ParentPath(), FileB.Filename()));
 
         EXPECT_FALSE(uuidA.IsNull());
         EXPECT_FALSE(uuidB.IsNull());
         EXPECT_NE(uuidA, uuidB);
     }
+
+    TEST_F(UuidManagerTests, GetUuid_CorruptedFile_Fails)
+    {
+        static constexpr AZ::IO::FixedMaxPath TestFile = "c:/somepath/mockfile.txt";
+        static constexpr AZ::IO::FixedMaxPath MetadataFile = TestFile.Native() + AzToolsFramework::MetadataManager::MetadataFileExtension;
+
+        MakeFile(TestFile);
+
+        auto uuid = m_uuidInterface->GetUuid(AssetProcessor::SourceAssetReference(TestFile));
+        auto result = AZ::Utils::ReadFile<AZStd::string>(MetadataFile.Native());
+
+        ASSERT_TRUE(result.IsSuccess());
```

Comment:
question: Just checking, does this need to be `ASSERT_TRUE` instead of `EXPECT_TRUE`

Reply:
suggestion: Potentially splitting the test with comment sections e.g. "Given, When, Then" may help readability here

Reply:
Yes because the result value is read on the next line which will fail if result has not succeeded.  I've updated some of the tests to have some more comments about what's going on